ase/clapplugin.cc

```diff
@@ -398,10 +406,33 @@ class ClapAudioWrapper : public AudioProcessor {
     output_events_.resize (0);
   }
   void
+  update_transportinfo()
+  {
+    const auto& trans = transport();
+    const auto& tick_sig = trans.tick_sig;
+    transportinfo.flags = CLAP_TRANSPORT_HAS_TEMPO |
+                          CLAP_TRANSPORT_HAS_BEATS_TIMELINE |
+                          CLAP_TRANSPORT_HAS_SECONDS_TIMELINE |
+                          CLAP_TRANSPORT_HAS_TIME_SIGNATURE;
+    transportinfo.tempo      = tick_sig.bpm();
+    transportinfo.tsig_num   = tick_sig.beats_per_bar();
+    transportinfo.tsig_denom = tick_sig.beat_unit();
+    double beat_pos = 0, sec_pos = 0;
+    if (trans.running())
+      {
+        transportinfo.flags |= CLAP_TRANSPORT_IS_PLAYING;
+        beat_pos = trans.current_tick_d * (1.0 / TRANSPORT_PPQN);
+        sec_pos  = trans.current_seconds + trans.current_minutes * 60;
+      }
+    transportinfo.song_pos_beats = llrint (beat_pos * CLAP_BEATTIME_FACTOR);
+    transportinfo.song_pos_seconds = llrint (sec_pos * CLAP_SECTIME_FACTOR);
+  }
```

Comment:
I'd like to see *all* transportinfo fields be setup here, even if those are just set to 0.
Also, I really prefer the fields to be setup in the order they are declared in `events.h:clap_event_transport`, so a missing initialization is easy to spot.

The loop_* fields should probably be 0 for now. I'm not sure how `tempo_inc` is meant to be used though. For the moment I guess it should be 0 as long as we don't support BPM changes?

Reply:
Ok, I now list all the fields in the order clap events.h uses. Btw, I also added values for the clap transport fields `bar_start` and `bar_number` now. The `loop_*` fields are set to 0. Indeed `tempo_inc` only makes sense if the tempo can be changed on the master track.